students/paul_jurek/lesson03/mailroom.py

```diff
@@ -0,0 +1,126 @@
+
+
+def send_thank_you(fullname):
+    """prints thank you message to terminal for donation"""
+    print(f'Thank you {fullname} for your generous donation!')
+
+
+def report():
+    """handles process for main screens report selection
+
+    If the user (you) selected “Create a Report”, print a list of your donors,
+    sorted by total historical donation amount.
+    Include Donor Name, total donated, number of donations and average
+    donation amount as values in each row. You do not need to print out all
+    their donations, just the summary info.
+    Using string formatting, format the output rows as nicely as possible.
+    The end result should be tabular (values in each column should align
+    with those above and below)
+    After printing this report, return to the original prompt.
+    At any point, the user should be able to quit their current task and
+    return to the original prompt.
+    From the original prompt, the user should be able to quit the script
+    cleanly.
+    Your report should look something like this:
+    Donor Name                | Total Given | Num Gifts | Average Gift
+    ------------------------------------------------------------------
+    William Gates, III         $  653784.49           2  $   326892.24
+    Mark Zuckerberg            $   16396.10           3  $     5465.37
+    Jeff Bezos                 $     877.33           1  $      877.33
+    Paul Allen                 $     708.42           3  $      236.14
+    """
+    print(f"{'Donor Name':<26}|{'Total Given':^15}|{'Num Gifts':^11}|{'Average Gift':^15}")
+    print('-'*70)
+    donor_stats = []
+    for donor in donors.keys():
+        donor_stats.append(summarize_donor(donor))
+        donor_stats.sort(key=lambda tup: tup[1], reverse=True)
+    
+    for summary in donor_stats:
+        print(f"{summary[0]:<26} ${summary[1]:>13.2f}  {summary[2]:>10}  ${summary[3]:>14.2f}")
+
+
+def summarize_donor(donor_name):
+    """generates donor summary
+    args:
+        donor_name: donor name matching key from donors
+    returns
+        tuple with fields (donor name, total given, num gifts, average gift)"""
+    total_given = sum(donors[donor_name])
+    num_gifts = len(donors[donor_name])
+    average_gift = total_given/num_gifts
+    
+    return(donor_name, total_given, num_gifts, average_gift)
+
+
+if __name__ == '__main__':
+    # initial placeholder for input
+    user_input = None
+
+    # run until user specifies to get out
+    while user_input != 'quit':
+        user_input = input('Options:\n'
+                           '\tSend a Thank You\n'
+                           '\tCreate a Report\n'
+                           '\tquit\n'
+                           'Please input option: ')
+
+        # cleans up user input to make more robust.
+        user_input = user_input.lower().strip()
+
+        if user_input == 'send a thank you':
+            thank_you()
+        elif user_input == 'create a report':
+            report()
```

Comment:
would recommend using integer values for selections, having users type in full strings are prone to errors 🙂